### PR TITLE
Add traceFlags settings to enable mdsd tracing in AMA via public settings

### PR DESCRIPTION
**Pull request by AaronRM**
Add traceFlags settings to enable mdsd tracing in AMA via public settings.

### CODE REVIEW

File: AzureMonitorAgent/agent.py [hunk 1]
```diff
@@ -417,13 +417,18 @@ def enable():
         ("azuremonitoragentmgr", False)
     ])
 
+    # Set traceFlags in publicSettings to enable mdsd tracing. For example, the EventIngest flag can be enabled via "traceFlags": "0x2"
+    flags = ""
+    if public_settings is not None and "traceFlags" in public_settings:
+        flags = "-T {} ".format(public_settings.get("traceFlags"))
```
Comment (simathih) [inline AzureMonitorAgent/agent.py:423]: shouldn't it be {0}?

File: AzureMonitorAgent/agent.py [hunk 2]
```diff
@@ -417,13 +417,18 @@ def enable():
...
+
     # Use an Ordered Dictionary to ensure MDSD_OPTIONS (and other dependent variables) are written after their dependencies
     default_configs = OrderedDict([
         ("MDSD_CONFIG_DIR", "/etc/opt/microsoft/azuremonitoragent"),
         ("MDSD_LOG_DIR", "/var/opt/microsoft/azuremonitoragent/log"),
         ("MDSD_ROLE_PREFIX", "/run/azuremonitoragent/default"),
         ("MDSD_SPOOL_DIRECTORY", "/var/opt/microsoft/azuremonitoragent"),
-        ("MDSD_OPTIONS", "\"-A -c /etc/opt/microsoft/azuremonitoragent/mdsd.xml -d -r $MDSD_ROLE_PREFIX -S $MDSD_SPOOL_DIRECTORY/eh -L $MDSD_SPOOL_DIRECTORY/events\""),
+        ("MDSD_OPTIONS", "\"{}-A -c /etc/opt/microsoft/azuremonitoragent/mdsd.xml -d -r $MDSD_ROLE_PREFIX -S $MDSD_SPOOL_DIRECTORY/eh -L $MDSD_SPOOL_DIRECTORY/events\"".format(flags)),
```
Comment (simathih) [inline AzureMonitorAgent/agent.py:431]: shouldn't it be {0}?

File: AzureMonitorAgent/agent.py (hunk 1 above)
Comment (AaronRM) [reply AzureMonitorAgent/agent.py:423]: [The index is not necessary on Python 3.1+](https://docs.python.org/3/library/string.html#formatstrings). "The positional argument specifiers can be omitted for [str.format()](https://docs.python.org/3/library/stdtypes.html#str.format), so '{} {}'.format(a, b) is equivalent to '{0} {1}'.format(a, b)."

File: AzureMonitorAgent/agent.py (hunk 2 above)
Comment (AaronRM) [reply AzureMonitorAgent/agent.py:431]: See previous comment.

**Review by simathih — APPROVED** [review]

**Review by NarineM — APPROVED** [review]

**Pull request merged**